dataland-frontend/tests/component/components/forms/CreateEuTaxonomyForNonFinancials.cy.ts

```diff
@@ -0,0 +1,298 @@
+      { force: true },
+    );
+    cy.get(".p-dialog-content").should("contain.text", "already uploaded");
+    cy.get(".p-dialog-header-close").click();
+    cy.get(`[data-test="${TEST_PDF_FILE_NAME}ToUploadContainer"]`).should("have.length", 1);
+  }
+
+  /**
+   * Adds a report to upload and removes it again afterwards checking that no dialog regarding a duplicate file name
+   * is wrongly triggered and that the file is correctly removed.
+   */
+  function checkExistingFilenameDialogDidNotBreakSubsequentSelection(): void {
+    const reportNameA = TEST_PDF_FILE_NAME;
+    const reportNameB = `${TEST_PDF_FILE_NAME}2`;
+    uploadDocuments.selectFile(reportNameB, "referencedReports");
+
+    cy.get(".p-dialog-content").should("not.exist");
+    uploadDocuments.validateReportToUploadIsListedInFileSelectorAndHasInfoForm(reportNameA);
+    uploadDocuments.validateReportToUploadIsListedInFileSelectorAndHasInfoForm(reportNameB);
+
+    uploadDocuments.removeReportFromSelectionForUpload(TEST_PDF_FILE_NAME);
+
+    uploadDocuments.validateReportIsNotAlreadyUploadedOrSelectedForUpload(reportNameA);
+    uploadDocuments.validateReportToUploadIsListedInFileSelectorAndHasInfoForm(reportNameB);
+    uploadDocuments.validateNumberOfReportsSelectedForUpload(1);
+
+    uploadDocuments.removeReportFromSelectionForUpload(reportNameB);
+
+    uploadDocuments.validateReportIsNotAlreadyUploadedOrSelectedForUpload(reportNameB);
+    uploadDocuments.validateNumberOfReportsSelectedForUpload(0);
+  }
+
+  /**
+   * On the eu taxonomy for non-financial services edit page, this method checks that submission is denied
+   * if a report is not referenced
+   */
+  function checkThatFilesMustBeReferenced(): void {
+    uploadDocuments.fillAllFormsOfReportsSelectedForUpload();
+    submitButton.clickButton();
+    cy.get('[data-test="failedUploadMessage"]').should("exist").should("contain.text", "test-report");
+    cy.get('[data-test="failedUploadMessage"]')
+      .should("exist")
+      .should("satisfy", (element: JQuery<HTMLElement>) => {
+        const expectedStrings = ["Not all uploaded reports are used", "test-report"];
+        const elementText = element.text();
+        return expectedStrings.every((expectedString) => elementText.includes(expectedString));
+      });
+  }
+
+  /**
+   * this method fills and checks the general section
+   * @param reports the name of the reports that are uploaded
+   */
+  function fillAndValidateGeneralSection(reports: string[]): void {
+    cy.get('[data-test="fiscalYearEnd"] button').should("have.class", "p-datepicker-trigger").click();
+    cy.get("div.p-datepicker").find('button[aria-label="Next Month"]').click();
+    cy.get("div.p-datepicker").find('span:contains("11")').click();
+    cy.get('input[name="fiscalYearEnd"]').invoke("val").should("contain", "11");
+    cy.get('input[name="fiscalYearDeviation"][value="Deviation"]').check();
+    cy.get('div[data-test="submitSideBar"] li:last a').click();
```

Comment:
I find this hard to understand. Any chance we can make this more expressive?

Reply:
I have removed this line as it just clicks on the submit sidebar, but does not check if the scrolling takes place and the sidebar is another component so in my opinion it should not be part of this component test.